src/Sylius/Bundle/AssociationBundle/DependencyInjection/Configuration.php

```diff
@@ -0,0 +1,126 @@
+    }
+
+    /**
+     * @param ArrayNodeDefinition $node
+     */
+    private function addResourcesSection(ArrayNodeDefinition $node)
+    {
+        $node
+            ->children()
+                ->arrayNode('resources')
+                    ->useAttributeAsKey('name')
+                    ->prototype('array')
+                        ->children()
+                            ->scalarNode('subject')->isRequired()->end()
+                            ->arrayNode('association')
+                                ->isRequired()
+                                ->addDefaultsIfNotSet()
+                                ->children()
+                                    ->arrayNode('classes')
+                                        ->addDefaultsIfNotSet()
+                                        ->children()
+                                            ->scalarNode('model')->isRequired()->end()
+                                            ->scalarNode('controller')->defaultValue(ResourceController::class)->end()
+                                            ->scalarNode('factory')->defaultValue(Factory::class)->end()
+                                            ->arrayNode('form')
+                                                ->addDefaultsIfNotSet()
+                                                ->children()
+                                                    ->scalarNode('default')->defaultValue(AssociationType::class)->end()
+                                                ->end()
+                                            ->end()
+                                        ->end()
+                                    ->end()
+                                    ->arrayNode('validation_groups')
+                                        ->addDefaultsIfNotSet()
+                                        ->children()
+                                            ->arrayNode('default')
+                                                ->prototype('scalar')->end()
+                                                ->defaultValue(array('sylius'))
+                                            ->end()
+                                        ->end()
+                                    ->end()
+                                ->end()
+                            ->end()
+                            ->arrayNode('association_type')
+                                ->addDefaultsIfNotSet()
+                                ->children()
+                                    ->arrayNode('classes')
+                                        ->addDefaultsIfNotSet()
+                                        ->children()
+                                            ->scalarNode('model')->defaultValue(AssociationTypeModel::class)->end()
+                                            ->scalarNode('controller')->defaultValue(ResourceController::class)->end()
+                                            ->scalarNode('factory')->defaultValue(Factory::class)->end()
```

Comment:
Add `->scalarNode('repository')->cannotBeEmpty()->end()` to allow using own repository implementations.